storybook/pagesmodel.cpp

```diff
@@ -0,0 +1,90 @@
+#include "pagesmodel.h"
+
+#include <QRegularExpression>
+#include <QDir>
+
+namespace {
+const auto categoryFirst QStringLiteral("_");
+const auto categoryUncategorized QStringLiteral("Uncategorized");
+}
+
+PagesModel::PagesModel(const QString &path, QObject *parent)
+    : QAbstractListModel{parent}
+{
+    QDir dir(path);
+    dir.setFilter(QDir::Files);
+
+    static QRegularExpression fileNameRegex(
+                QRegularExpression::anchoredPattern("(.*)Page\\.qml"));
+    static QRegularExpression categoryRegex(
+                "^//(\\s)*category:(.+)$", QRegularExpression::MultilineOption);
+
+    const QFileInfoList files = dir.entryInfoList();
+
+    std::for_each(files.begin(), files.end(), [this] (auto &fileInfo) {
+        QString fileName = fileInfo.fileName();
+        QRegularExpressionMatch fileNameMatch = fileNameRegex.match(fileName);
+
+        if (!fileNameMatch.hasMatch())
+            return;
+
+        QFile file(fileInfo.filePath());
+        file.open(QIODevice::ReadOnly);
+        QByteArray content = file.readAll();
+
+        QRegularExpressionMatch categoryMatch = categoryRegex.match(content);
+        QString category = categoryMatch.hasMatch()
+                ? categoryMatch.captured(2).trimmed() : categoryUncategorized;
+
+        QString title = fileNameMatch.captured(1);
+        m_items << PagesModelItem { title, category };
+    });
+
+    std::sort(m_items.begin(), m_items.end(),
```

Comment:
I'd remove _any_ sorting from here, you can do that in QML using the `SortFilterProxyModel`

Reply:
Ok, I agree it's not ideal place for that. Moved to qml - despite much slower (ExpressionSorter), code will be a bit better organised.